emails/views.py

```diff
@@ -396,6 +398,14 @@ def _sns_message(message_json):
     _record_receipt_verdicts(receipt, 'active_alias')
     incr_if_enabled('email_for_active_address', 1)
 
+    # if address is blocking list emails, and email is from list, early return
+    email_is_from_list = _check_email_from_list(mail['headers'])
+    if address and address.block_list_emails and email_is_from_list:
```

Comment:
Nit: This may need the S3 deletion logic from PR #1543. But then every early return may need it...

https://github.com/mozilla/fx-private-relay/blob/9e154ab4ffb938b948bee295972dd81140ef857b/emails/views.py#L470-L473

Reply:
Sounds good. If we do a follow-up PR to clean that can you add the work for early return while checking the email being relayed is a reply on line 393?